### DISCUSSION TITLE
How to take screenshots of the AR mode (picture of the model and the environment) in the webXR?

### DISCUSSION THREAD

**Discussion by gabrielgaldinomagalhaes** [google/model-viewer#4256] [Q&A]
I'm trying to code a way to allow the users to save screenshots of the AR experience in the webXR.

I tried somethings as < canvas > that doesn't get the model viewer "viewport" and toBlob but it only get the model and doensn't get the environment that the model is in. None of them worked.

What would be the best way to make that?
I just want to allow the user to take pictures inside the AR experience (screenshots)

**Comment by elalish** [comment]
Probably the right way to do this is as a feature in MV. I believe you'll need to request camera permission to get the background, then composite our render on top. It hasn't been enough of a priority for us to tackle yet, but we'd be very supportive of a PR. Feel free to start a draft and ask questions in the PR if you're having trouble and we'll help.

**Comment by danbaechtold** [comment]
Looking for this as well.. are there any news on this, even if just experimental? Thx!

**Reply by elalish** [reply]
Alas no - interested in making a PR?

**Comment by k1pp0** [comment]
Hi there!

I've been personally interested in this feature, so I experimented with AR screenshot capture in WebXR mode and got it working:
→ https://k1pp0.github.io/model-viewer-webxr-capture/

Since this was built without modifying model-viewer's core, the implementation is more complex than it needs to be. A native implementation would make it much simpler and more accessible.

This feature has also been discussed in #1481.
  
Is there still intention to officially support this? If so, I'd be happy to contribute a PR.

Thanks!
